lib/file_beam/middleware/https_redirect.ex

```diff
@@ -0,0 +1,65 @@
+defmodule FileBeam.Middleware.HttpsRedirect do
+  alias Raxx.Server
+  alias Raxx.Middleware
+  alias Raxx.Request
+
+  @behaviour Middleware
+
+  @moduledoc """
+  Redirects all http requests to https, except for when on localhost
+  """
+
+  @impl Middleware
+  def process_head(%Request{scheme: :https} = request, _config, inner_server) do
+    # all good, do nothing
+    {parts, inner_server} = Server.handle_head(inner_server, request)
+    {parts, :state, inner_server}
+  end
+
+  def process_head(%Request{authority: "localhost" <> _} = request, _config, inner_server) do
```

Comment:
If your extracting this, might want to make this configurable. I was building a similar thing but actually want to check https locally.

Reply:
Yeah, I just didn't get to making it configureable, I figured this was good enough for now.

With localhost you can use both http and https, you just won't get redirected.